### ISSUE TITLE
import! ignores columns parameter and tries to update all columns

### ISSUE THREAD

**Issue by unchartedworks** [zdennis/activerecord-import#593]
`     Level.import! changed_levels, on_duplicate_key_update: columns`
Level is an subclass of ActiveRecord

**Expected Result:**
import! should only update the specified columns in columns parameter but not all of columns in Level.
**Actual Result:**
Columns parameters are ignored, as a result, all of columns in Level are updated in spite of that they are not in the columns.

It's a new bug which could be reproduced in 1.0.0, and I reverted to the older version 0.27.0, this issue is gone.

**Version:** 1.0.0

**Comment by jkowens** [comment]
Can you try version 0.28 as well?

**Comment by unchartedworks** [comment]
Fortunately a problematic seed file prevented importing data, otherwise my database would be ruined. The last working version was version 0.27.0 before I upgraded gems. I can't really try version 0.28 because I can't afford losing data.

**Comment by jkowens** [comment]
I created a basic test script and it appears to be working as expected, only updating specified columns. I'm thinking you may have discovered a scenario not covered in our tests. Can you provide me with a basic code snippet that would reproduce the issue? Even better would be a sample repo I could clone 👍

**Comment by unchartedworks** [comment]
```
       # Bulk-import changed levels.
      update_columns = Level.columns.map(&:name).map(&:to_sym).
        reject {|column| %i(id name created_at).include? column}
      Level.import! changed_levels, on_duplicate_key_update: update_columns
```

**Comment by jkowens** [comment]
My test basically covers that. I'm not sure why I can't reproduce your error. How did you determine it was attempting to update all columns?

**Comment by unchartedworks** [comment]
Caused by:
Mysql2::Error: Incorrect datetime value: ‘2015-10-16T22:39:29.000Z’ for column ‘created_at’ at row 1
/dashboard/lib/level_loader.rb:64:in `block in import_levels’

Column ‘created_at’ should not be updated if import! works as expected.

**Comment by jkowens** [comment]
I don't think that error necessarily means on duplicate key update was going to update that field. I'd have to check, but I believe MySQL validates data types before considering uniqueness constraints. What I don't really understand is why you got the error `Mysql2::Error: Incorrect datetime value: ‘2015-10-16T22:39:29.000Z’`. Do you know why that error happened?

**Comment by unchartedworks** [comment]
I am sorry for the late reply. When I use version 0.27.0, I don't get this error.

**Comment by jkowens** [comment]
I'm starting to wonder if the root problem is v1 is not properly parsing timestamps. There was another issue reported about timestamps recently (#583).

**Comment by jkowens** [comment]
How is `created_at` being set on these objects? Are you manually setting it or are you fetching already created records from the database?

**Comment by unchartedworks** [comment]
I set it manually when an object is created. And these columns should not be updated.

**Comment by jkowens** [comment]
Fix released in v1.0.1
